client/finality-grandpa/src/finality_proof.rs

```diff
@@ -42,55 +39,49 @@
 use std::sync::Arc;
 use log::trace;
 
-use sp_blockchain::{Backend as BlockchainBackend, Error as ClientError, Result as ClientResult};
-use sc_client_api::{
-	backend::Backend, StorageProof,
-	light::{FetchChecker, RemoteReadRequest},
-	StorageProvider, ProofProvider,
+use sp_blockchain::{
+	Backend as BlockchainBackend, Error as ClientError, Result as ClientResult,
 };
+use sc_client_api::{backend::Backend, StorageProvider, ProofProvider};
 use parity_scale_codec::{Encode, Decode};
 use finality_grandpa::BlockNumberOps;
 use sp_runtime::{
 	Justification, generic::BlockId,
 	traits::{NumberFor, Block as BlockT, Header as HeaderT, One},
 };
 use sp_core::storage::StorageKey;
-use sc_telemetry::{telemetry, CONSENSUS_INFO};
 use sp_finality_grandpa::{AuthorityId, AuthorityList, VersionedAuthorityList, GRANDPA_AUTHORITIES_KEY};
 
 use crate::justification::GrandpaJustification;
 use crate::VoterSet;
+use crate::SharedAuthoritySet;
 
-/// Maximum number of fragments that we want to return in a single prove_finality call.
-const MAX_FRAGMENTS_IN_PROOF: usize = 8;
-
-/// GRANDPA authority set related methods for the finality proof provider.
 pub trait AuthoritySetForFinalityProver<Block: BlockT>: Send + Sync {
```

Comment:
Do we still need this trait and impls?

Reply:
I've removed  `StorageAndProofProvider`, but kept `AuthoritySetForFinalityProver` since it's useful for the tests